cmd/xl-storage.go

```diff
@@ -1789,42 +1795,43 @@ func (s *xlStorage) CheckFile(ctx context.Context, volume string, path string) e
 		return err
 	}
 
-	// Stat a volume entry.
-	_, err = os.Stat(pathJoin(volumeDir, path))
-	if err != nil {
-		if osIsNotExist(err) {
+	var checkFile func(p string) error
+	checkFile = func(p string) error {
+		if p == "." || p == SlashSeparator {
 			return errPathNotFound
 		}
-		return err
-	}
 
-	filePath := pathJoin(volumeDir, path, xlStorageFormatFile)
-	if err = checkPathLength(filePath); err != nil {
-		return err
-	}
+		filePath := pathJoin(volumeDir, p, xlStorageFormatFile)
+		if err := checkPathLength(filePath); err != nil {
+			return err
+		}
 
-	filePathOld := pathJoin(volumeDir, path, xlStorageFormatFileV1)
-	if err = checkPathLength(filePathOld); err != nil {
-		return err
-	}
+		filePathOld := pathJoin(volumeDir, p, xlStorageFormatFileV1)
```

Comment:
Just wondering. Is there no way that the `*xlStorage` can know that it was created after the rename. Here it would cut the number of stats in half and a lot of other places could benefit?

Reply:
Actually wait we can do @klauspost we need to check if it's a legacy backend or not.

If it's not then we don't need to have this check.